Cargo.lock

```diff
@@ -0,0 +1,105 @@
+# This file is automatically @generated by Cargo.
+# It is not intended for manual editing.
+version = 3
+
+[[package]]
+name = "calendar-duration"
+version = "1.0.0"
+dependencies = [
+ "time",
+]
+
+[[package]]
+name = "deranged"
```

Comment:
This is a [PoC](https://github.com/jhpratt/deranged) but appears to be well maintained.